### ISSUE TITLE
[BUG] CC: inconsistent/buggy example messages behavior esp. AI Studio

### ISSUE THREAD

**Issue by cloak1505** [SillyTavern/SillyTavern#2997]
### Environment

🪟 Windows

### System

Firefox

### Version

1.12.6 'staging' (8986cb0)/(725cfdeb)

### Describe the problem

**Solo Chat**

AI Studio: When "Use system prompt" is disabled, names in example messages are gone and replaced by "example_user" and "example_assistant". Though not too big of an issue, this doesn't feel right since I'd expect them to be the same.
![image](https://github.com/user-attachments/assets/cd829d13-98ba-4450-8b08-8f88be762197)

Claude (system prompt on/off) and Cohere are OK. They prefix character names.

**Group Chat** - stuff happens.

AI Studio "Use system prompt" disabled - this is a single user message.
* Example messages containing only {{char}} disappear while [Example Chat] still shows.
* For most examples, a single "example_user" or "example_assistant" appears, but somehow Nino's examples alternate between them including double newlines.
* Ichika has misplaced {{user}} sent to top when {{char}} is first.
<details>
<summary>Screenshot of group chat examples (no system prompt)</summary>

![image](https://github.com/user-attachments/assets/ba037e0e-3d2f-42bb-a6f6-ce4756515c50)

</details>

AI Studio "Use system prompt" enabled - this is a single `system_instruction`.
* Mostly error free except for...
* This time Nino has a misplaced {{user}} sent to top when {{char}} is first.
* Ichika's first example is the one with double newlines.
<details>
<summary>Screenshot of group chat examples (in system prompt)</summary>

![image](https://github.com/user-attachments/assets/09adc5cf-ec07-455f-afdb-d9c9fc12de38)

</details>

Claude - looks good except the same misplaced {{user}} as above.
<details>
<summary>Screenshot of Claude system prompt</summary>

![image](https://github.com/user-attachments/assets/f75b2179-4036-4b81-b38b-59991fad75eb)

</details>

I just now updated (2 days apart, probably no difference) and reran the group chat request with the Assistant test example messages. Ichika's messages are back but the {{char}}-only message at bottom is gone.

<details>
<summary>Screenshot of AI Studio group chat with more example messages (no system prompt)</summary>

![image](https://github.com/user-attachments/assets/1c8b6e3e-582b-4776-b113-98df49e4ccd4)

</details>

Edit: Tried again, all 3 {{char}}-only messages are blank under Cohere and Google (sys on/off).

2024-10-27: Discovered the cause of the inconsistency.
> I realize that the "active character" will have their {{char}}-only examples show, but not the others.

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by cloak1505** [comment]
What uses name property aside from OpenAI? Even for OpenAI, is it necessary? If it's good, why don't the user use "Completion Object" option to apply object names to chat history too?

OpenRouter: For non-OpenAI models, OR will move example_user/assistant into the message as prefixes:
```
[Example Chat]
example_user: Example message for {{user}}.
example_assistant: Example message for {{char}}.
```
ST includes names in Group Chat, which for non-OpenAI models will end up like this (ugly):	
```
[Example Chat]
example_user: User: Example message for {{user}}.
example_assistant: Char: Example message for {{char}}.
```
What if we treated OpenRouter like other APIs for example messages (use character names) but enable name property when "Completion Object" names behavior is selected? This will also bypass issue #2981 as long as user doesn't select "Completion Object". I suppose this is easier said than done since you can see the object name thing appear in Prompt Inspector under other APIs though the actual request uses name prefixes. Smells like spaghetti code, idk.

**Comment by cloak1505** [comment]
Prefixes under Gemini are fixed.

What's left is the mysteriously disappearing {{char}}-only example messages in group chat with joined cards. I realize that the "active character" will have their {{char}}-only examples show, but not the others. Applies to all APIs. Not affected by order of group member list. Hope this helps.

**Comment by cloak1505** [comment]
This monstrosity is completely fixed by #3191.
